Upgrade SDK, Firebase, null_safety

torstenek opened the pull request:
Upgraded dependencies to SDK >=2.12.0 and Firebase_core ^1.0.0 along with required transitive dependencies. Works with SDK 2.13.0-77.0.dev

awhitford commented on lib/src/firebase_image.dart:
```diff
@@ -35,18 +35,19 @@ class FirebaseImage extends ImageProvider<FirebaseImage> {
     this.maxSizeBytes = 2500 * 1000, // 2.5MB
     this.cacheRefreshStrategy = CacheRefreshStrategy.BY_METADATA_DATE,
-    this.firebaseApp,
+    required this.firebaseApp,
```
This breaking change should be documented in the change log and/or readme.

awhitford replied:
Is this _breaking change_ necessary?

torstenek replied:
Agreed. firebaseApp is no longer required.

awhitford commented:
The top-level `pubspec.lock` file should be _removed_ (and added to `.gitignore`).  It should not be committed for a library, only for an app.

See https://github.com/toptal/gitignore/blob/2d8160c7f607e142929d2ba213620a638e50603f/templates/Flutter.gitignore#L16

awhitford commented:
Actually, I think PR #30 is slightly better because it leverages the `late` keyword.

torstenek committed:
Upgrade SDK, Firebase, null_safety

Upgraded dependencies to SDK >=2.12.0 and Firebase_core ^1.0.0 along with required transitive dependencies. Works with SDK 2.13.0-77.0.dev

torstenek commented:
Using the late keyword is better. Changed it. Sorry for not spotting this in an earlier PR.

torstenek commented:
I'll leave this PR here since this is the version I'm using, and in the event that it can be useful. I suppose it would be obviated by a merge of PR #30. As with PR #30 it would probably benefit from further testing.

mattreid1 commented:
Just merged PR #30. Thanks for your help though!

torstenek commented:
NP 👍

On Tue, 9 Mar 2021 at 15:43, Matt Reid <notifications@github.com> wrote:

> Just merged PR #30 <https://github.com/mattreid1/firebase_image/pull/30>.
> Thanks for your help though!
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/mattreid1/firebase_image/pull/32#issuecomment-793987952>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AABKUWOTMXGGAJFFVCLMAVTTCYXZNANCNFSM4YX2KAVQ>
> .
>
